tasks.py

```diff
@@ -182,12 +217,92 @@ def format(c):
 
 
 @task
-def security_check(c):
+def security_check(context: Context) -> None:
     """Run pip-autid on dependencies"""
-    c.run(
+    context.run(
         """
         pipenv requirements > requirements.txt && \
         pipenv run pip-audit -r requirements.txt && \
         rm -rf requirements.txt
         """
     )
+
+
+@task
+def setup_pre_commit_hooks(context: Context) -> None:
+    """Setup pre-commit hook to automate check before git commit and git push"""
+    context.run("git init")
+    context.run(
+        "pipenv run pre-commit install -t pre-commit & "
+        "pipenv run pre-commit install -t pre-push & "
+        "pipenv run pre-commit install -t commit-msg &"
+        "pipenv run pre-commit autoupdate"
+    )
+
+
+@task
+def run_pre_commit(context: Context) -> None:
+    """Run pre-commit on all-files"""
+    context.run("pipenv run pre-commit run --all-files")
+
+
+def _ask_multiple_inputs_question(prompt: str, break_symbol: str = "!") -> str:
+    questionary.print(f'{prompt} Enter "{break_symbol}" to finish"')
+    answers = []
+    while (answer := questionary.text("", qmark="->").ask()) != break_symbol:
+        answers.append(answer)
+    return ", ".join(answer)
+
+
+def _validate_datetime(datetime_str: str) -> bool:
+    try:
+        datetime.strptime(datetime_str, "%Y-%m-%d %H:%M:%S")
+        return True
+    except ValueError:
+        return False
+
+
+@task
+def create_post(context: Context) -> None:
+    """Create a new post with required metadata."""
+    questionary.print("Create a new post", style="bold")
+
+    answers = questionary.form(
+        title=questionary.text(
+            "Title of the post: ", validate=lambda answer: answer != ""
+        ),
+        date=questionary.text(
+            "Date: ",
+            default=datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+            validate=_validate_datetime,
+        ),
+        category=questionary.select("Select a category:", choices=AVAILABLE_CATEGORIES),
+    ).ask()
+    tags = _ask_multiple_inputs_question("Type any number of tags.")
+    authors = _ask_multiple_inputs_question("Specify any number of authors.")
```

Comment:
I've never know there is such a great tool for asking problems. Please take my knee 🙇🏻

Reply:
it's used by commitizen-tools haha